src/main/java/org/pac4j/dropwizard/Pac4jConfiguration.java

```diff
@@ -1,7 +1,172 @@
 package org.pac4j.dropwizard;
 
-import io.dropwizard.Configuration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-public interface Pac4jConfiguration<T extends Configuration> {
-    Pac4jFactory getPac4jFactory(T configuration);
+import org.pac4j.core.authorization.authorizer.Authorizer;
+import org.pac4j.core.authorization.generator.AuthorizationGenerator;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.matching.Matcher;
+import org.pac4j.jax.rs.filter.SecurityFilter;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An interface defining getters for parameters required to configure a {@link SecurityFilter}.
+ */
+public class Pac4jConfiguration {
```

Comment:
In other Dropwizard libraries, [classes like this](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-db/src/main/java/io/dropwizard/db/DataSourceFactory.java) which encapsulate configuration options are termed "-Factory" rather than "-Configuration". In which case, this class would be `Pac4jFactory`, as before.